python/tskit/tables.py

```diff
@@ -1663,6 +1663,28 @@ def simplify(
             reduce_to_site_topology=reduce_to_site_topology,
             keep_unary=keep_unary)
 
+    def map_ancestors(self, samples, ancestors):
+        """
+        Returns an :class:`EdgeTable` instance displaying all relationships between
+        the nodes supplied in `samples` and `ancestors`.
+        Relationships between different nodes in `ancestors` are also shown.
+        The supplied nodes must be a subset of the node IDs in the tree sequence.
+        Other than this, the procedure is flexible; overlap between the lists of samples
+        and ancestors is permitted, and the nodes need not be ordered by time.
+        Node IDs are not changed in the output.
+        If none of the nodes in `ancestors` are ancestral to `samples` anywhere in
+        the tree sequence, an empty table will be returned.
```

Comment:
This is great. Could you add another short paragraph saying precisely how the EdgeTable displays the relationships, @gtsambos? For instance, say that the IDs in the edge table refer to the original tree sequence; and also say when exactly an edge appears in the output edge table.

Reply:
> For each sequence of edges that joins a sample node to an ancestral node

Hm, this is still a big ambiguous: if a tree has a -> b -> c, where a and b are ancestors and c is a sample, then there's a path from c to a but the output doesn't have a row for it.

How about, erm,
```
Returns an :class:`EdgeTable` describing how ``samples`` are descended from ``ancestors`` (and how ``ancestors`` are descended from each other). Each row ``parent, child, left, right`` in the output table indicates that ``child`` (which is either in ``samples`` or in ``ancestors``) has inherited the segment ``[left, right)`` from ``parent`` (which is in ``ancestors``) more recently than from any other node in ``ancestors``. Every segment of the genome on which each of ``samples`` has inherited from any of ``ancestors`` is described in this way, as well as any segment of the genomes of ``ancestors`` that has been inherited by any of ``samples``.  The node IDs in ``parent`` and ``child`` refer to the node table in the original tree sequence. In other words, the tree constructed at a position with the output edge table has the property that the parent of any node in ``samples``, if there is one, is the most recent ancestor of that node at that position out of all possible ancestors in ``ancestors``, and the same is true for any node in ``ancestors`` that is ancestral to any of ``samples``.
```

Is this right? Particularly about including edges with ancestors in them iff they are ancestral to samples?

This made me think of another case: what happens if a,b are samples and x,y are ancestors and there's a segment on which the relationship is x -> a -> y -> b?  Do we get
```
parent child
 x           a
 x           y
 y           b
```
...?

Reply:
Thanks - have updated the description!

> This made me think of another case: what happens if a,b are samples and x,y are ancestors and there's a segment on which the relationship is x -> a -> y -> b? Do we get ...

Yep, that's right. I've added a new test case showing this behaviour (`test_single_tree_ancestors_descend_from_samples`)

Reply:
(Maybe that test name should be changed, as it implies that the ancestors and samples are completely interchangeable which they actually aren't.) Eg. if in the above example you'd set samples = [a] and ancestors = [x, y], you'd only get the first row of that output.

Reply:
Agh, just realised I should have looked at your example a bit more carefully.
You don't get this
```
parent child
 x           a
 x           y
 y           b
```

but this
```
parent child
 x           a
 a           y
 y           b
```

Reply:
Hmm, just thinking that x -> a -> b is another distinct case that I need a distinct test for.
The `samples` and `ancestors` are almost the same - the only difference is that an `ancestor` will *only* be retained if it has a relevant descendant, while a `sample` will *always* be retained.

Reply:
Hm, that means the documentation is not quite right. Can you perhaps fix it up?

Reply:
Sorry, I didn't realise there was still unresolved issues here. We can fix this with a follow up PR though, right?

Reply:
Yep, I'll open up another pull request with these small changes once I've made them.

Reply:
thanks! I could do it, but I'm more certain that you know exactly what it does than I do.